milli/src/vector/hf.rs

```diff
@@ -70,7 +70,7 @@ impl std::fmt::Debug for Embedder {
 
 impl Embedder {
     pub fn new(options: EmbedderOptions) -> std::result::Result<Self, NewEmbedderError> {
-        let device = candle_core::Device::Cpu;
+        let device = candle_core::Device::cuda_if_available(0).unwrap();
```

Comment:
Does it panic if CUDA is unavailable?

Reply:
🤩 Good catch, the behavior was a bit subtle here:

1. Without the cuda feature, it would always just return `Ok(Device::Cpu)`, so no risk in packing in this case
2. With the cuda feature, it would panic if the CUDA support library could not be found or if the GPU doesn't support it for some reason.

I modified the code to avoid the panic and fallback to CPU in all cases (with a warning stating the issue), it should feel more robust now 👍